src/basic/hostname-util.c

```diff
@@ -9,22 +9,31 @@
 
 #include "alloc-util.h"
 #include "hostname-util.h"
+#include "os-util.h"
 #include "string-util.h"
 #include "strv.h"
 
-const char* get_fallback_hostname(void) {
-        const char *e;
+char* get_fallback_hostname(void) {
+        int r;
 
-        e = secure_getenv("SYSTEMD_FALLBACK_HOSTNAME");
+        const char *e = secure_getenv("SYSTEMD_FALLBACK_HOSTNAME");
         if (e) {
                 if (!hostname_is_valid(e, 0))
                         log_debug("Invalid hostname in $SYSTEMD_FALLBACK_HOSTNAME, ignoring.");
                 else
-                        return e; /* The pointer is valid as long as a getenv(), putenv(), setenv(), or
-                                   * unsetenv() are not called. */
+                        return strdup(e);
         }
 
-        return FALLBACK_HOSTNAME;
+        _cleanup_free_ char *f = NULL;
+        r = parse_os_release(NULL, "FALLBACK_HOSTNAME", &f);
```

Comment:
why no use the ID field as fallback here?

Reply:
This creates a breaking change in behaviour in some sense (so far, the change to the default/fallback/whatever hostname were opt-in). By using ID for this, we'll change the hostname on existing systems. And some people might be unhappy about this, because of network-configuration issues, but also because of a perceived privacy leak.

Dunno, I see the appeal, but I'm not sure if it's worth the potential fallout.